### PR TITLE
Add Fortran bindings

### PR DESCRIPTION
**Pull request by astrofrog**
Closes https://github.com/conda-forge/hdf5-feedstock/issues/11

Continued version of #14

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -32,22 +32,25 @@ requirements:
     - python    # [win]
     - cmake >=3.1
     - zlib 1.2*
+    - gcc  # [not win]
```
Comment (gillins) [inline recipe/meta.yaml]: I thought we only needed `gcc` for Linux and not OSX. Is this right @jakirkham ?

Comment (jakirkham) [inline recipe/meta.yaml]: Not when it comes to Fortran. That is the only way we have to build Fortran on OS X. We will need to make sure the `clang` stuff stays entact. We don't want this to start shipping `libgcc` on OS X if we can avoid it.

Comment (gillins) [inline recipe/meta.yaml]: Yes agree about `libfortran` but `gcc` should have the `linux` selector instead? Is that right?

Comment (jakirkham) [inline recipe/meta.yaml]: Oh you mean the line below.

File: recipe/meta.yaml [hunk 2]
```diff
@@ -32,22 +32,25 @@ requirements:
...
+    - libgfortran  # [not win]
```
Comment (jakirkham) [inline recipe/meta.yaml]: This should be dropped.

File: recipe/meta.yaml [hunk 3]
```diff
@@ -32,22 +32,25 @@ requirements:
...
   run:
     - zlib 1.2*
+    - libgcc  # [not win]
```
Comment (jakirkham) [inline recipe/meta.yaml]: This should also be dropped.

File: recipe/meta.yaml (hunk 1 above)
Comment (jakirkham) [inline recipe/meta.yaml]: Sorry I misread. Though we do have to do something about that line too.

Unfortunately we need `gcc` on OS X too if we want to have access to `gfortran`. Mac doesn't provide a  `gfortran` compiler. I believe they use to, but that was on 10.4. Fortunately, the work we did getting `clang` to be the compiler and `libc++` used should actually help us here. As a result, we should only need to ship `libgfortran` with this on OS X. I just went through a similar ordeal with `scipy`. So, am pretty confident that this should work. We still want to review this carefully nonetheless.

File: recipe/meta.yaml (hunk 2 above)
Comment (astrofrog) [inline recipe/meta.yaml]: @jakirkham - in https://github.com/conda-forge/hdf5-feedstock/pull/28#discussion_r71445033 (which is more recent that this comment), it sounded like you were saying this line should be kept - do you mean this one should be removed, or the similar line in the runtime dependencies?

Comment (jakirkham) [inline recipe/meta.yaml]: Ah, do something meant remove it. Sorry that was really ambiguous. So, yes, let's drop this.

We may need to tinker with this stuff later. I had a lot of fun doing this with SciPy. We may need to do a similar exploration here. Fortunately for you I have done this once now. 😄

Comment (astrofrog) [inline recipe/meta.yaml]: Sounds good :) Just to check, do we then also drop it as a runtime dependency?

Comment (jakirkham) [inline recipe/meta.yaml]: Drop the `libgcc` run dependency, but keep `libgfortran`. That's ideally how we want to ship this. We may need some more tweaks to get it there.

**Comment by jakirkham** [comment]
Yeah, actually, I'm going to propose the exact same thing we did with `scipy` (which has worked out quite well. Please add this [`yum_requirements.txt`]( https://raw.githubusercontent.com/conda-forge/scipy-feedstock/ec005340e3e79436f59fb859759869163af128dc/recipe/yum_requirements.txt ) to the `recipe` directory. Then make these other changes that follow.

File: recipe/meta.yaml [hunk 4]
```diff
@@ -34,22 +34,24 @@ requirements:
     - libtool        # [unix]
     - cmake >=3.1
     - zlib 1.2*
+    - gcc  # [not win]
```
Comment (jakirkham) [inline recipe/meta.yaml]: Change `# [not win]` -> `# [osx]`.

File: recipe/meta.yaml [hunk 5]
```diff
@@ -34,22 +34,24 @@ requirements:
...
+    - libgfortran  # [not win]
```
Comment (jakirkham) [inline recipe/meta.yaml]: Change `# [not win]` -> `# [linux]`.

File: recipe/meta.yaml [hunk 6]
```diff
@@ -34,22 +34,24 @@ requirements:
...
   run:
     - zlib 1.2*
+    - libgfortran  # [not win]
```
Comment (jakirkham) [inline recipe/meta.yaml]: 👍

File: recipe/build.sh [hunk 7]
```diff
@@ -11,13 +11,16 @@ source activate "${CONDA_DEFAULT_ENV}"
 if [ "$(uname)" == "Darwin" ]
 then
     export CXX="${CXX} -stdlib=libc++"
+    export DYLD_FALLBACK_LIBRARY_PATH=$PREFIX/lib
 fi
```
Comment (jakirkham) [inline recipe/build.sh]: Please add `export LIBRARY_PATH="${PREFIX}/lib"`.

**Comment by jakirkham** [comment]
These changes were needed with SciPy to ensure that we did not need to ship `libstdc++`. The same story exists here where we build support for C++ bindings and don't want to ship `libstdc++` on Linux. It is a bit hacky, but has been found to be effective.

File: recipe/build.sh [hunk 8]
```diff
@@ -12,6 +12,7 @@ if [ "$(uname)" == "Darwin" ]
 then
     export CXX="${CXX} -stdlib=libc++"
     export DYLD_FALLBACK_LIBRARY_PATH=$PREFIX/lib
+    export LIBRARY_PATH="${PREFIX}/lib"
```
Comment (jakirkham) [inline recipe/build.sh]: Oh this should be for both platforms. In particular, we need this for Linux.

File: recipe/build.sh [hunk 9]
```diff
@@ -11,13 +11,17 @@ source activate "${CONDA_DEFAULT_ENV}"
 if [ "$(uname)" == "Darwin" ]
 then
     export CXX="${CXX} -stdlib=libc++"
+    export DYLD_FALLBACK_LIBRARY_PATH=$PREFIX/lib
+    export LIBRARY_PATH="${PREFIX}/lib"
```
Comment (jakirkham) [inline recipe/build.sh]: Oh this last line should be for both platforms. In particular, we need this for Linux.

**Comment by jakirkham** [comment]
Also forgot to mention please do this locally in the top level of this repo.

* `conda install -n root -c conda-forge conda-smithy=0.10.5`.
* `conda smithy rerender`
* `git add .`
* `git commit -m "Re-rendered with conda-smithy 0.10.5."`
* `git push`

This should let us use the `yum_requirements.txt`.

**Comment by astrofrog** [comment]
@jakirkham - thanks for the tips! Strangely, re-rendering led to the Python 3 builds being dropped from AppVeyor, so I did not include that change in the latest commit. Should I have done?

File: .travis.yml [hunk 10]
```diff
@@ -26,7 +26,7 @@ install:
 
       conda config --set show_channel_urls true
       conda update --yes conda
-      conda install --yes conda-build jinja2 anaconda-client
+      conda install --yes conda-build=1.20.0 jinja2 anaconda-client
```
Comment (jakirkham) [inline .travis.yml]: I know it generated this, but we should not include it. See this issue ( https://github.com/conda-forge/hdf5-feedstock/issues/23 ) for background as to why we removed it.

File: recipe/build.sh (hunk 7 above)
Comment (jakirkham) [inline recipe/build.sh]: Done below.

**Comment by jakirkham** [comment]
> @jakirkham - thanks for the tips!

Of course.

> Strangely, re-rendering led to the Python 3 builds being dropped from AppVeyor, so I did not include that change in the latest commit. Should I have done?

Nope, you were right not to. I think you are running into this issue ( https://github.com/SciTools/conda-build-all/issues/49 ). If you are able to create a reproducer and add it to that issue, that would be amazing. I've had such trouble trying to get anything that would be reproducible.

**Comment by astrofrog** [comment]
Looks like everything is passing! :)

**Comment by jakirkham** [comment]
Thanks @astrofrog. Could I ask that we add some test code here? I would like to see a couple test programs (it could be copied from HDF5's examples for all I care). Something for Fortran, C, and C++. I expect this will work, but I want to be very sure that it does. These tests can be Unix only. Building in the test phase for Windows has proved very problematic and we have not changed that build AFAICT.

**Comment by astrofrog** [comment]
@jakirkham - no problem, I'll add some test code today!

**Comment by astrofrog** [comment]
I've added some test cases that I copied from the HDF5 examples - I picked one that includes compression etc. to really test things. I checked and included the HDF5 license to make sure this is acceptable.

File: recipe/meta.yaml [hunk 11]
```diff
@@ -34,22 +34,24 @@ requirements:
...
 
 test:
   requires:
-    - python 2.7.*  # [win and py27]
-    - python 3.4.*  # [win and py34]
-    - python 3.5.*  # [win and py35]
+    - python {{ environ['PY_VER'] + '*' }}  # [win]
```
Comment (jakirkham) [inline recipe/meta.yaml]: Need to add `gcc` for OS X only.

**Comment by jakirkham** [comment]
Once we get this one worked out, I'd like to test it out some locally. I really want to be sure we don't accidentally break `hdf5` or end up with some weird linking mess. Unfortunately, I don't think the the CI test will be sufficient alone to do this. Though the tests that you added, @astrofrog, should be very helpful in this process. I hope everyone can understand my desire to take this slow.

File: recipe/meta.yaml (hunk 11 above)
Comment (astrofrog) [inline recipe/meta.yaml]: I thought this was already the case:

```
    - gcc  # [osx]
```

I think GitHub messed up the lines for the comments, @jakirkham can you clarify which line you were referring to?

Comment (jakirkham) [inline recipe/meta.yaml]: Ah, but I'm suggesting we make it a test requirement. Travis CI is complaining we don't have `gfortran`. So this should fix it.

**Comment by astrofrog** [comment]
@jakirkham - absolutely! I wonder if it would be worth pushing my branch to a prerelease branch in the feedstock so that we can all test out the builds by going to a particular label?

**Comment by astrofrog** [comment]
Ok, so the MacOS X build is now failing with:

```
dyld: Library not loaded: @rpath/./libgfortran.3.dylib
  Referenced from: /Users/travis/miniconda3/conda-bld/test-tmp_dir/./h5_cmprss
  Reason: image not found
/Users/travis/miniconda3/conda-bld/test-tmp_dir/run_test.sh: line 17: 77628 Trace/BPT trap: 5       ./h5_cmprss
```

I'm assuming this should **not** be fixed with the fallback path, because we can't expect users to actually do this too. Does anyone know why the libgfortran library wouldn't get picked up at this point? (I'm pretty sure it's installed since it's in the runtime dependencies).

**Comment by jakirkham** [comment]
Are we setting the library search path?

**Comment by astrofrog** [comment]
@jakirkham - sorry, I wasn't clear: what I meant was that during the build, we set ``DYLD_FALLBACK_LIBRARY_PATH`` to make sure that libgfortran gets picked up, but in the testing phase we don't. I don't think we want to add it during the testing phase because we can't expect that the users also have this set. But I'm not sure how else we can solve this?

**Comment by jakirkham** [comment]
However, this is exactly the same problem. The executable we are trying to run has not had `install_name_tool` run on it to set the rpath. We have two options.

1. Set `DYLD_FALLBACK_LIBRARY_PATH`.
2. Add this compiler flag for the linker `-Wl,-rpath,$PREFIX/lib`.

**Comment by astrofrog** [comment]
@jakirkham - just to make sure I understand, will the situation be different for the user? (can we expect that ``install_name_tool`` will get run in their case?)

**Comment by jakirkham** [comment]
I guess I'm having trouble seeing a use case where a user would not just build with `conda-build`. I know we are for more tests, but that seems a bit different. Could you help me understand what use cases you have in mind where this will be an issue, @astrofrog?

**Comment by astrofrog** [comment]
@jakirkham - ah sorry, I should have made clear, this is failing during the compilation of the test program. The build of the conda package succeeded. So if a user installs the conda package for hdf5 and tries to do:

```
h5fc mycode.f90
```

then they might run into this error. I'm worried about implementing workarounds based on setting ``DYLD_FALLBACK_LIBRARY_PATH`` because I don't think we want users to do this just so ``h5fc`` works correctly.

**Comment by astrofrog** [comment]
@jakirkham - I think this is actually an issue with the gcc conda package: https://github.com/conda/conda/issues/3241 - I'm not sure what the best approach is moving forward, whether we should just disable the testing for MacOS X so that we can merge this PR, or if we need to wait for the gcc conda package to be fixed?

**Comment by jakirkham** [comment]
Let's just set `DYLD_FALLBACK_LIBRARY_PATH` on OS X. I know it is suboptimal, but I'm having trouble coming up with a way around it ATM.

**Comment by jakirkham** [comment]
So I have made some proposed tweaks to `libgfortran` in PR ( https://github.com/conda-forge/libgfortran-feedstock/pull/3 ). It would be nice if you could test this out, @astrofrog (assuming you have a Mac). Though even if we do get all issues here sorted, we may still be blocked thanks to issue ( https://github.com/conda/conda-build/issues/1167 ), which seems to be causing us build issues with very trivial changes ( see PR https://github.com/conda-forge/hdf5-feedstock/pull/29 ).

**Comment by jakirkham** [comment]
Actually there was a `conda-build` release. Will restart CIs for those PRs to see if we still have issues.

File: recipe/run_test.sh [hunk 12]
```diff
@@ -0,0 +1,27 @@
...
+# the conda library path. Ideally this shouldn't be needed, but this is how the
+# gfortran compiler works in conda - see:
+#
+#   https://github.com/ContinuumIO/anaconda-issues/issues/739
+#
+# for more details.
+
+echo "Testing h5fc"
+h5fc -show
+h5fc -Wl,-rpath,${CONDA_PREFIX}/lib h5_cmprss.f90 -o h5_cmprss
```
Comment (jakirkham) [inline recipe/run_test.sh]: 👍

**Comment by astrofrog** [comment]
@jakirkham - thanks, I've now set DYLD_FALLBACK_LIBRARY_PATH here for now (note that the issue in gcc won't get fixed: https://github.com/ContinuumIO/anaconda-issues/issues/739). I'll try out the libgfortran in conda-forge/libgfortran-feedstock#3 this week.

**Comment by jakirkham** [comment]
Oh did `-Wl,-rpath,` not work? Honestly either way should be fine. Whichever is convenient really.

I'm really not sure if that change to the `libgfortran` package is going to be enough. We may need to do something similar here to what we did with [`openblas`]( https://github.com/conda-forge/openblas-feedstock/blob/18a1d3ac43a6f1050cdeb71c06fa96949f414ffd/recipe/build.sh#L24-L39 ).

**Comment by astrofrog** [comment]
@jakirkham - I just wanted to check both cases, and both indeed pass on Travis. I have no preference, so let me know if you want me to change it back. If this PR passes overall, can we merge it and continue the ``libgfortran`` experiments separately?

**Comment by jakirkham** [comment]
Could I ask you to merge PR ( https://github.com/conda-forge/hdf5-feedstock/pull/29 ) into the changes here? They are trivial version pinning changes that shouldn't have any negative effect on this, but it would be nice to have them taken care of.

**Comment by jakirkham** [comment]
> If this PR passes overall, can we merge it and continue the `libgfortran` experiments separately?

This seems like a reasonable request to me. I think this is in good shape and would be happy to get it out there for people to use. However, I know I'm not going to have lots of time to debug issues we may find in this package if they do arise. So I have a bit of an unconventional request.

Is it alright if put the packages under a label like `test` instead of `main` on the `conda-forge` channel? This way everyone has an opportunity to chose to try out these changes and if they find issues they can report them. However, if the package does end up being broken, we have not forced it on to users that don't need it. After say a week or two (or some other agreeable time frame), we can move it to `main` and see how that goes. What do you think?

**Comment by astrofrog** [comment]
> Is it alright if put the packages under a label like test instead of main on the conda-forge channel? 

Absolutely - how should we do this?

**Comment by astrofrog** [comment]
@jakirkham - thinking out loud - it would be cool if the latest build from a PR got uploaded with a label such as e.g. pr-28 or similar. Then anyone could easily test the latest build from a PR. Should I open an issue with this idea, or is it definitely not doable

**Comment by jakirkham** [comment]
So this is my thought process. We can create a `test_fortran` branch on the main feedstock and configure it to upload to a label in a PR that also includes all of your changes here. Was thinking of using the label `test`, but if you have a preference for a different label we can do that. How does this sound?

**Comment by jakirkham** [comment]
> thinking out loud - it would be cool if the latest build from a PR got uploaded with a label such as e.g. pr-28 or similar. Then anyone could easily test the latest build from a PR. Should I open an issue with this idea, or is it definitely not doable

In terms of scaling up the process? Yeah, this does seem like an interesting idea worth exploring. 👍

If you could open an issue on the webpage repo outlining this a bit more, that would be great. Being able to test pre-releases is something conda-forge would really benefit from. Many of the issues I can remember from the last few months would have been much more easily resolved with this strategy.

**Comment by jakirkham** [comment]
> So this is my thought process. We can create a `test_fortran` branch on the main feedstock and configure it to upload to a label in a PR that also includes all of your changes here. Was thinking of using the label `test`, but if you have a preference for a different label we can do that. How does this sound?

Should add that we will then merge this PR into `master` once we are comfortable with having it out there.

**Comment by jakirkham** [comment]
I have put the changes in this [branch]( https://github.com/jakirkham/hdf5-feedstock/tree/config_test_fortran ) on my fork. Was planning on PR-ing that against the `test_fortran` branch. It basically absorbs the content of this PR and makes some changes so as to upload the packages under the `test` label on the `conda-forge` channel.

**Comment by jakirkham** [comment]
So I have created PR ( https://github.com/conda-forge/hdf5-feedstock/pull/31 ) that should do exactly as described. Please let me know if you think this is reasonable, @astrofrog.

**Comment by astrofrog** [comment]
@jakirkham - that all sounds great. Feel free to merge #31 once it passes, and I'll test out the builds.

**Comment by astrofrog** [comment]
> If you could open an issue on the webpage repo outlining this a bit more, that would be great. Being able to test pre-releases is something conda-forge would really benefit from. Many of the issues I can remember from the last few months would have been much more easily resolved with this strategy.

Ok, see https://github.com/conda-forge/conda-forge.github.io/issues/204

**Comment by jakirkham** [comment]
One other thing, would you be willing to be a maintainer of `hdf5`, @astrofrog? Given all the work you have put into getting Fortran to work here, it seems like you know quite a bit about what is involved and what still needs to be done.

**Comment by astrofrog** [comment]
@jakirkham - yes, I'd be happy to. Should I add myself in this PR?

**Comment by jakirkham** [comment]
Sure, please do. I'll update the test PR afterwards.

**Comment by astrofrog** [comment]
@jakirkham - done!

**Comment by jakirkham** [comment]
Thanks @astrofrog. Updated the testing PR accordingly. 😄

**Comment by astrofrog** [comment]
Thinking about this again, I'm not sure if it would actually be that risky to simply merge this - all this does is adds a compilation flag for the library that adds the fortran bindings, but the C bindings should be unchanged. So even if there are still issues to iron out with the fortran, it will be no worse than currently (where there is no support at all).

**Comment by ocefpaf** [comment]
There is no risk here. I am using your changes for a while in a local build
for NOAA with great success.

+1

**Comment by jakirkham** [comment]
I'm reluctant to put this in `main` as I really don't have time to debug `hdf5` issues should they arise. That probably won't change for a couple of weeks. It was very upsetting to me to have it rebuild everything around `openblas` when a seemingly safe PR caused problems. It resulted in me wasting a day rebuilding stuff when I really did not have the time to do so. We are still dealing with the consequences of that PR now.

If you would like to give us a 👍 on releasing it under the `test` label as mentioned in PR ( https://github.com/conda-forge/hdf5-feedstock/pull/32 ) and PR ( https://github.com/conda-forge/hdf5-feedstock/pull/31 ), @ocefpaf I would happily go ahead with those. We can then reassess this in a few weeks time.

**Comment by astrofrog** [comment]
Just to say that this seems to work nicely for https://github.com/conda-forge/staged-recipes/pull/1483 and I've also been using it locally with success. I think @ocefpaf has also been using the version from the test label successfully. What is the path forward to go ahead and merge this? Are there some specific packages for which we want to try updating to the version in the test label?

**Comment by ocefpaf** [comment]
> I think @ocefpaf has also been using the version from the test label successfully. 

Not the test label. In fact I am using this for months now, even before that the test label, in a local build system we created for NOAA to circumvent some weird decision in conda-forge.

**Comment by astrofrog** [comment]
@ocefpaf - ok, thanks for the clarification - but you are using the recipe as it is in this PR?

**Comment by ocefpaf** [comment]
> @ocefpaf - ok, thanks for the clarification - but you are using the recipe as it is in this PR?

Exactly.

**Comment by astrofrog** [comment]
(just rebased)

**Comment by jakirkham** [comment]
> What is the path forward to go ahead and merge this? Are there some specific packages for which we want to try updating to the version in the test label?

Trying to use this with `h5py`, `pytables`, and possibly some other HDF5 libraries in conda-forge with this package would be good. The easiest thing would be to run their test suites, but if you have any creative ideas, I would be all ears. I'd also like to know if we are going to need to rebuild things.

**Comment by astrofrog** [comment]
I did some tests with HDF5 from the test label on MacOS X. The environment I am using has:
```
hdf5                      1.8.17                        3    conda-forge/label/test
ca-certificates           2016.8.31                     0    conda-forge
certifi                   2016.8.31                py35_0    conda-forge
h5py                      2.6.0               np111py35_6    conda-forge
libgfortran               3.0.0                         0    conda-forge
ncurses                   5.9                           9    conda-forge
numexpr                   2.6.1               np111py35_0    conda-forge
openssl                   1.0.2h                        2    conda-forge
pip                       8.1.2                    py35_0    conda-forge
pytables                  3.3.0               np111py35_0    conda-forge
python                    3.5.2                         3    conda-forge
readline                  6.2                           0    conda-forge
setuptools                26.1.1                   py35_0    conda-forge
six                       1.10.0                   py35_0    conda-forge
sqlite                    3.13.0                        1    conda-forge
tk                        8.5.19                        0    conda-forge
wheel                     0.29.0                   py35_0    conda-forge
xz                        5.2.2                         0    conda-forge
zlib                      1.2.8                         3    conda-forge
mkl                       11.3.3                        0  
numpy                     1.11.1                   py35_0  
```

The h5py tests pass:

```
In [3]: h5py.tests.run_tests()
....................................................x.......................s.......................................x....................................s........ss..............................................................................................ssssss.................................................................x....x..x..................x.....x......................................ssssssssss............
----------------------------------------------------------------------
Ran 423 tests in 1.839s

OK (skipped=20, expected failures=7)
Out[3]: <unittest.runner.TextTestResult run=423 errors=0 failures=0>
```

and likewise the pytables tests pass:

```
In [1]: import tables
tables.
In [2]: tables.test()
-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
PyTables version:    3.3.0
HDF5 version:        1.8.17
NumPy version:       1.11.1
Numexpr version:     2.6.1 (not using Intel's VML/MKL)
Zlib version:        1.2.8 (in Python interpreter)
BZIP2 version:       1.0.6 (6-Sept-2010)
Blosc version:       1.11.1 (2016-09-03)
Blosc compressors:   blosclz (1.0.5), lz4 (1.7.2), lz4hc (1.7.2), snappy (1.1.1), zlib (1.2.8), zstd (1.0.0)
Blosc filters:       shuffle, bitshuffle
Python version:      3.5.2 | packaged by conda-forge | (default, Sep  8 2016, 14:36:38) 
[GCC 4.2.1 Compatible Apple LLVM 6.0 (clang-600.0.54)]
Platform:            Darwin-15.6.0-x86_64-i386-64bit
Byte-ordering:       little
Detected cores:      2
Default encoding:    utf-8
Default FS encoding: utf-8
Default locale:      (en_US, UTF-8)
-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
Performing only a light (yet comprehensive) subset of the test suite.
If you want a more complete test, try passing the --heavy flag to this script
(or set the 'heavy' parameter in case you are using tables.test() call).
The whole suite will take more than 4 hours to complete on a relatively
modern CPU and around 512 MB of main memory.
-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
/Users/tom/miniconda3/envs/test-hdf5/lib/python3.5/site-packages/tables/filters.py:330: FiltersWarning: compression library ``lzo`` is not available; using ``zlib`` instead
  % (complib, default_complib), FiltersWarning)
version: 1.11.1
./Users/tom/miniconda3/envs/test-hdf5/lib/python3.5/site-packages/tables/path.py:112: NaturalNameWarning: object name is not a valid Python identifier: 'a:b'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though
  NaturalNameWarning)
/Users/tom/miniconda3/envs/test-hdf5/lib/python3.5/site-packages/tables/path.py:112: NaturalNameWarning: object name is not a valid Python identifier: '/a/b'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though
  NaturalNameWarning)
/Users/tom/miniconda3/envs/test-hdf5/lib/python3.5/site-packages/tables/path.py:112: NaturalNameWarning: object name is not a valid Python identifier: '.'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though
  NaturalNameWarning)
..................../Users/tom/miniconda3/envs/test-hdf5/lib/python3.5/site-packages/tables/path.py:112: NaturalNameWarning: object name is not a valid Python identifier: 'a:b'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though
  NaturalNameWarning)
/Users/tom/miniconda3/envs/test-hdf5/lib/python3.5/site-packages/tables/path.py:112: NaturalNameWarning: object name is not a valid Python identifier: '/a/b'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though
  NaturalNameWarning)
/Users/tom/miniconda3/envs/test-hdf5/lib/python3.5/site-packages/tables/path.py:112: NaturalNameWarning: object name is not a valid Python identifier: '.'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though
  NaturalNameWarning)
..................../Users/tom/miniconda3/envs/test-hdf5/lib/python3.5/site-packages/tables/path.py:112: NaturalNameWarning: object name is not a valid Python identifier: 'a:b'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though
  NaturalNameWarning)
/Users/tom/miniconda3/envs/test-hdf5/lib/python3.5/site-packages/tables/path.py:112: NaturalNameWarning: object name is not a valid Python identifier: '/a/b'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though
  NaturalNameWarning)
/Users/tom/miniconda3/envs/test-hdf5/lib/python3.5/site-packages/tables/path.py:112: NaturalNameWarning: object name is not a valid Python identifier: '.'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though
  NaturalNameWarning)
..................../Users/tom/miniconda3/envs/test-hdf5/lib/python3.5/site-packages/tables/path.py:112: NaturalNameWarning: object name is not a valid Python identifier: 'a:b'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though
  NaturalNameWarning)
/Users/tom/miniconda3/envs/test-hdf5/lib/python3.5/site-packages/tables/path.py:112: NaturalNameWarning: object name is not a valid Python identifier: '/a/b'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though
  NaturalNameWarning)
/Users/tom/miniconda3/envs/test-hdf5/lib/python3.5/site-packages/tables/path.py:112: NaturalNameWarning: object name is not a valid Python identifier: '.'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though
  NaturalNameWarning)
..................../Users/tom/miniconda3/envs/test-hdf5/lib/python3.5/site-packages/tables/path.py:112: NaturalNameWarning: object name is not a valid Python identifier: 'a:b'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though
  NaturalNameWarning)
/Users/tom/miniconda3/envs/test-hdf5/lib/python3.5/site-packages/tables/path.py:112: NaturalNameWarning: object name is not a valid Python identifier: '/a/b'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though
  NaturalNameWarning)
/Users/tom/miniconda3/envs/test-hdf5/lib/python3.5/site-packages/tables/path.py:112: NaturalNameWarning: object name is not a valid Python identifier: '.'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though
  NaturalNameWarning)
..................../Users/tom/miniconda3/envs/test-hdf5/lib/python3.5/site-packages/tables/path.py:112: NaturalNameWarning: object name is not a valid Python identifier: 'a:b'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though
  NaturalNameWarning)
/Users/tom/miniconda3/envs/test-hdf5/lib/python3.5/site-packages/tables/path.py:112: NaturalNameWarning: object name is not a valid Python identifier: '/a/b'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though
  NaturalNameWarning)
/Users/tom/miniconda3/envs/test-hdf5/lib/python3.5/site-packages/tables/path.py:112: NaturalNameWarning: object name is not a valid Python identifier: '.'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though
  NaturalNameWarning)
....................................................................................................ss..................................................................................................................................................................................................................................................................ssss...........................s.........................................................................................................................sssss.................................................................................................................................................................ssssssssssssssssssssssssssss............................................................................................................................................................................................................................................................................................................................................................................ssssssssssssssssssssssssssssssssssssssssssssssssssssssss.......................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................ssssss...................................................................................................................................................................................................................................................................................................................................................................................................................................................sssssss........................................................ssssssssssssss...................................................................................................................................................................................................ssssssssss...........................................................................ssssssssss.................................................................sssss...............sssss......................................................................................................................................................................................ssssss..............................................................................................................................................................................................................s...............................................................................................................................................................................................................................................................................................s...s.......................................................................................................................................................................................................................................................................................................................................................<string>:1: RuntimeWarning: invalid value encountered in arcsin
.........................<string>:1: RuntimeWarning: invalid value encountered in arccos
...................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................ssssssssssssssssssss............................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................/Users/tom/miniconda3/envs/test-hdf5/lib/python3.5/site-packages/tables/nodes/filenode.py:170: UserWarning: host PyTables file is already closed!
  warnings.warn("host PyTables file is already closed!")
......
----------------------------------------------------------------------
Ran 6173 tests in 629.514s

OK (skipped=182)
```

On Linux, the h5py tests pass:


```
In [1]: import h5py

In [2]: h5py.tests.run_tests()
....................................................x.......................s.......................................x....................................s........ss..............................................................................................ssssss.................................................................x....x..x..................x.....x......................................ssssssssss............
----------------------------------------------------------------------
Ran 423 tests in 0.610s

OK (skipped=20, expected failures=7)
Out[2]: <unittest.runner.TextTestResult run=423 errors=0 failures=0>
```

and I'm currently running the 'heavy' pytables test suite.

**Comment by astrofrog** [comment]
The full test suite for pytables ran (in 3 hours!) with no issues:

```
Ran 67114 tests in 11490.133s

OK (skipped=43)
```

These packages didn't need re-building or anything, I just used the latest conda-forge versions.

@jakirkham @pelson - any other tests you would like me to do before we can merge this?

**Comment by pelson** [comment]
From my perspective, I see no blockers, but I haven't been that close to the discussion. I think @astrofrog's testing looks pretty extensive - @jakirkham, do you have any remaining concerns that @astrofrog can address, or are you OK to let this fly?

**Comment by jakirkham** [comment]
So was PyTables tested on Linux too, @astrofrog? Am a little unclear. Not worried about the heavy test suite on both, but just confirming that the standard test suite works on both OS X and Linux.

**Comment by astrofrog** [comment]
@jakirkham - yes, I ran the normal pytables suite on Mac and the heavy one on Linux, and the h5py test suite on bo

**Comment by astrofrog** [comment]
@jakirkham - should I increase the build number btw? or does that happen automatically?

**Comment by jakirkham** [comment]
Yep, please bump the build number and then will merge.

**Comment by astrofrog** [comment]
@jakirkham - I think this is ready to go - thanks!

**Comment by jakirkham** [comment]
Thanks @astrofrog.

**Comment by jakirkham** [comment]
They changed the website in the last couple hours so the source download is failing. 😒 

When you have a chance, could you please add a PR to correct this, @astrofrog?

**Comment by astrofrog** [comment]
@jakirkham - done in https://github.com/conda-forge/hdf5-feedstock/pull/42

**Comment by jakirkham** [comment]
Thanks for doing that, @astrofrog.

**Pull request merged**